### PR TITLE
Added AntlrTokenFilter

### PR DESCRIPTION
**Pull request by tomidelucca**
Before submitting a PR, please check that:
 - [ ] The PR is submitted against `master`. The PMD team will merge back to support branches as needed.
 - [ ] `./mvnw clean verify` passes. This will [build](https://github.com/pmd/pmd/blob/master/BUILDING.md) and test PMD, execute PMD and checkstyle rules. [Check this for more info](https://github.com/pmd/pmd/blob/master/CONTRIBUTING.md#code-style)

**PR Description:**

### CODE REVIEW

File: pmd-core/src/main/java/net/sourceforge/pmd/cpd/AntlrTokenizer.java [hunk 1]
```diff
@@ -23,18 +23,13 @@
     public void tokenize(final SourceCode sourceCode, final Tokens tokenEntries) {
 
         AntlrTokenManager tokenManager = getLexerForSource(sourceCode);
+        AntlrTokenFilter tokenFilter = getTokenFilter(tokenManager);
```
Comment (matifraga) [inline pmd-core/src/main/java/net/sourceforge/pmd/cpd/AntlrTokenizer.java]: Raro que no pidan finals

File: pmd-core/src/main/java/net/sourceforge/pmd/cpd/AntlrTokenizer.java [hunk 2]
```diff
@@ -47,8 +42,17 @@ public void tokenize(final SourceCode sourceCode, final Tokens tokenEntries) {
...
+    protected AntlrTokenFilter getTokenFilter(AntlrTokenManager tokenManager) {
+        return new AntlrTokenFilter(tokenManager);
+    }
+
     /* default */ static CharStream getCharStreamFromSourceCode(final SourceCode sourceCode) {
         StringBuilder buffer = sourceCode.getCodeBuffer();
         return CharStreams.fromString(buffer.toString());
     }
+
+    private void processToken(Tokens tokenEntries, String fileName, AntlrToken token) {
```
Comment (matifraga) [inline pmd-core/src/main/java/net/sourceforge/pmd/cpd/AntlrTokenizer.java]: finals?

File: pmd-core/src/main/java/net/sourceforge/pmd/cpd/token/BaseTokenFilter.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    /**
+     * Extension point for subclasses to analyze all tokens (before filtering)
+     * and update internal status to decide on custom discard rules.
+     *
+     * @param currentToken The token to be analyzed
+     * @see #isLanguageSpecificDiscarding()
+     */
+    protected void analyzeToken(final T currentToken) {
+        // noop
+    }
```
Comment (matifraga) [inline pmd-core/src/main/java/net/sourceforge/pmd/cpd/token/BaseTokenFilter.java]: Habría que revisar con juanma si esto queda abstract, o si queda no-op en las impls

**Review by matifraga — APPROVED** [review]

File: pmd-core/src/main/java/net/sourceforge/pmd/cpd/token/BaseTokenFilter.java (hunk 3 above)
Comment (lsoncini) [reply pmd-core/src/main/java/net/sourceforge/pmd/cpd/token/BaseTokenFilter.java]: Coincido

**Review by lsoncini — APPROVED** [review]

**Pull request closed without merging**